contracts/src/v0.8/ccip/test/pools/GHO/GhoTokenPoolRemote.t.sol

```diff
@@ -0,0 +1,557 @@
+    vm.expectEmit();
+    emit ChainConfigured(s_remoteChainSelector, newOutboundConfig, newInboundConfig);
+
+    changePrank(AAVE_DAO);
+    s_pool.setChainRateLimiterConfig(s_remoteChainSelector, newOutboundConfig, newInboundConfig);
+
+    uint256 expectedTokens = RateLimiter._min(newOutboundConfig.capacity, oldOutboundTokens);
+
+    RateLimiter.TokenBucket memory bucket = s_pool.getCurrentOutboundRateLimiterState(s_remoteChainSelector);
+    assertEq(bucket.capacity, newOutboundConfig.capacity);
+    assertEq(bucket.rate, newOutboundConfig.rate);
+    assertEq(bucket.tokens, expectedTokens);
+    assertEq(bucket.lastUpdated, newTime);
+
+    expectedTokens = RateLimiter._min(newInboundConfig.capacity, oldInboundTokens);
+
+    bucket = s_pool.getCurrentInboundRateLimiterState(s_remoteChainSelector);
+    assertEq(bucket.capacity, newInboundConfig.capacity);
+    assertEq(bucket.rate, newInboundConfig.rate);
+    assertEq(bucket.tokens, expectedTokens);
+    assertEq(bucket.lastUpdated, newTime);
+  }
+
+  function testOnlyOwnerOrRateLimitAdminSuccess() public {
+    address rateLimiterAdmin = address(28973509103597907);
+
+    changePrank(AAVE_DAO);
+    s_pool.setRateLimitAdmin(rateLimiterAdmin);
+
+    changePrank(rateLimiterAdmin);
+
+    s_pool.setChainRateLimiterConfig(
+      s_remoteChainSelector,
+      _getOutboundRateLimiterConfig(),
+      _getInboundRateLimiterConfig()
+    );
+
+    changePrank(AAVE_DAO);
+
+    s_pool.setChainRateLimiterConfig(
+      s_remoteChainSelector,
+      _getOutboundRateLimiterConfig(),
+      _getInboundRateLimiterConfig()
+    );
+  }
+
+  // Reverts
+
+  function testOnlyOwnerReverts() public {
+    changePrank(STRANGER);
+
+    vm.expectRevert(abi.encodeWithSelector(UpgradeableTokenPool.Unauthorized.selector, STRANGER));
+    s_pool.setChainRateLimiterConfig(
+      s_remoteChainSelector,
+      _getOutboundRateLimiterConfig(),
+      _getInboundRateLimiterConfig()
+    );
+  }
+
+  function testNonExistentChainReverts() public {
+    uint64 wrongChainSelector = 9084102894;
+
+    vm.expectRevert(abi.encodeWithSelector(UpgradeableTokenPool.NonExistentChain.selector, wrongChainSelector));
+    changePrank(AAVE_DAO);
+    s_pool.setChainRateLimiterConfig(
+      wrongChainSelector,
+      _getOutboundRateLimiterConfig(),
+      _getInboundRateLimiterConfig()
+    );
+  }
+}
+
+contract GhoTokenPoolRemote_setRateLimitAdmin is GhoTokenPoolRemoteSetup {
+  error OnlyCallableByOwner();
+
+  function testSetRateLimitAdminSuccess() public {
+    assertEq(address(0), s_pool.getRateLimitAdmin());
+    changePrank(AAVE_DAO);
+    s_pool.setRateLimitAdmin(OWNER);
+    assertEq(OWNER, s_pool.getRateLimitAdmin());
+  }
+
+  // Reverts
+
+  function testSetRateLimitAdminReverts() public {
+    vm.startPrank(STRANGER);
+
+    vm.expectRevert(OnlyCallableByOwner.selector);
+    s_pool.setRateLimitAdmin(STRANGER);
+  }
+}
+
+contract GhoTokenPoolRemote_directMint is GhoTokenPoolRemoteSetup {
```

Comment:
My personal preference is having very basic unit tests for `directMint` and `directBurn`, and a separate test that shows how the migration should work (instead of mixing up both). I think it's clearer and more helpful

Reply:
sure yep, how's this? [4fb7885](https://github.com/aave/ccip/pull/20/commits/4fb7885378e498aa203f87e659d4ab55e9903865)

Reply:
It's great now!